invenio_records_resources/references/entity_resolvers/results.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Resolver for result items."""
+
+from invenio_access.permissions import system_identity
+
+from .base import EntityProxy, EntityResolver
+
+
+class ResultItemProxy(EntityProxy):
```

Comment:
```suggestion
class ServiceResultProxy(EntityProxy):
```

minor: Technically, we're resolving to `ResultItem`s, but if we follow the naming convention of other `EntityProxy`s, they refer to the underlying entity/API.